tools/rosbag/src/recorder.cpp

```diff
@@ -609,7 +609,18 @@ void Recorder::doRecordSnapshotter() {
             OutgoingMessage out = out_queue.queue->front();
             out_queue.queue->pop();
 
-            bag_.write(out.topic, out.time, *out.msg);
+            ros::Time write_stamp = out.time;
```

Comment:
Two things:

- The snapshotter was also missing the connection header getting written out
- The snapshotter was always writing the timestamp of the original message, even if that message was latched and published thousands of seconds ago (like with global costmaps). This made the snapshot bags hard to use for nav.

Reply:
A part of me is worried about nodes tripping over updated timestamps (e.g. if we're replaying multiple sequential bags), but I don't know any specific nodes that would have issues.

Reply:
How does that differ from this?

https://github.com/locusrobotics/ros_comm/blob/eae18e0e37871518bebcebecc68ea1aa329fe82b/tools/rosbag/src/recorder.cpp#L413-L418

Reply:
Oh, huh. That block is rewriting rosbag's record of when the message was received. I misread this section as applying to the actual message headers, but obviously that's not what's actually happening on second look.